pkg/deploy/deploy.go

```diff
@@ -87,65 +87,19 @@ func deploy(client dtclient.Client, apis api.APIs, em *entityMap, c *config.Conf
 
 	case config.SettingsType:
 		log.Info("\tDeploying config %s", c.Coordinate)
-		return deploySetting(client, properties, renderedConfig, c)
+		return deploySetting(clientSet.settings(), properties, renderedConfig, c)
 
 	case config.ClassicApiType:
-		log.Info("\tDeploying config %s", c.Coordinate)
-		return deployConfig(client, apis, em, properties, renderedConfig, c)
+		return deployConfig(clientSet.classic(), apis, em, properties, renderedConfig, c)
 
+	case config.AutomationType:
+		log.Info("\tDeploying config %s", c.Coordinate)
+		return deployAutomation(clientSet.automation(), properties, renderedConfig, c)
 	default:
 		return nil, []error{fmt.Errorf("unknown config-type (ID: %q)", c.Type.ID())}
 	}
 }
 
-func deployConfig(configClient dtclient.ConfigClient, apis api.APIs, entityMap *entityMap, properties parameter.Properties, renderedConfig string, conf *config.Config) (*parameter.ResolvedEntity, []error) {
-	t, ok := conf.Type.(config.ClassicApiType)
-	if !ok {
-		return &parameter.ResolvedEntity{}, []error{fmt.Errorf("config was not of expected type %q, but %q", config.ClassicApiTypeId, conf.Type.ID())}
-	}
-
-	apiToDeploy, found := apis[t.Api]
-	if !found {
-		return &parameter.ResolvedEntity{}, []error{fmt.Errorf("unknown api `%s`. this is most likely a bug", t.Api)}
-	}
-
-	var errors []error
-	configName, err := extractConfigName(conf, properties)
-	if err != nil {
-		errors = append(errors, err)
-	} else if entityMap.contains(apiToDeploy.ID, configName) && !apiToDeploy.NonUniqueName {
-		errors = append(errors, newConfigDeployErr(conf, fmt.Sprintf("duplicated config name `%s`", configName)))
-	}
-	if len(errors) > 0 {
-		return &parameter.ResolvedEntity{}, errors
-	}
-
-	if apiToDeploy.DeprecatedBy != "" {
-		log.Warn("API for \"%s\" is deprecated! Please consider migrating to \"%s\"!", apiToDeploy.ID, apiToDeploy.DeprecatedBy)
-	}
-
-	var entity dtclient.DynatraceEntity
-	if apiToDeploy.NonUniqueName {
-		entity, err = upsertNonUniqueNameConfig(configClient, apiToDeploy, conf, configName, renderedConfig)
-	} else {
-		entity, err = configClient.UpsertConfigByName(apiToDeploy, configName, []byte(renderedConfig))
-	}
-
-	if err != nil {
-		return &parameter.ResolvedEntity{}, []error{newConfigDeployErr(conf, err.Error())}
-	}
-
-	properties[config.IdParameter] = entity.Id
-	properties[config.NameParameter] = entity.Name
-
-	return &parameter.ResolvedEntity{
-		EntityName: entity.Name,
-		Coordinate: conf.Coordinate,
-		Properties: properties,
-		Skip:       false,
-	}, nil
-}
-
 func upsertNonUniqueNameConfig(client dtclient.ConfigClient, apiToDeploy api.API, conf *config.Config, configName string, renderedConfig string) (dtclient.DynatraceEntity, error) {
```

Comment:
this method is Config API / ClassicAPIType specific -> it should also move to the new deploy/classic.go file :)